edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/DeviceConnectivityManager.cs

```diff
@@ -85,6 +91,13 @@ public DeviceConnectivityManager(TimeSpan minConnectivityCheckFrequency, TimeSpa
             Events.Created(minConnectivityCheckFrequency, disconnectedCheckFrequency);
         }
 
+        public void SetConnectionManager(IConnectionManager connectionManager)
```

Comment:
The connection manager can't come through the constructor?

Reply:
Or I guess, can it come through the constructor now?

Reply:
No, we will have the circular dependency - ConnectionManager -> CloudConnectionProvider -> ConnectivityAwareClientProvider -> DeviceConnectivityManager -> ConnectionManager.